### PR TITLE
networksettings: Skip testing device channel change for EOL distro

### PR DESCRIPTION
**Pull request by sharsonia**

### CODE REVIEW

File: microsoft/testsuites/network/networksettings.py [hunk 1]
```diff
@@ -174,6 +174,11 @@ def validate_ringbuffer_settings_change(self, node: Node) -> None:
         priority=1,
     )
     def validate_device_channels_change(self, node: Node, log: Logger) -> None:
+        if isinstance(node.os, Ubuntu) and node.os.information.release <= "16.04":
+            raise SkippedException(
+                f"The distro {node.os.name} {node.os.information.version} is EOL."
+                "It might not support device channel change."
+            )
```
Comment (LiliDeng) [inline microsoft/testsuites/network/networksettings.py:187]: I didn't find a clear kernel version, which is after this kernel version, the case is pass, before this kernel version, the case is failed, probably it also relates with the vm size, please check.

Comment (sharsonia) [reply microsoft/testsuites/network/networksettings.py:187]: > I didn't find a clear kernel version, which is after this kernel version, the case is pass, before this kernel version, the case is failed, probably it also relates with the vm size, please check.

Let me test it out with some other VM sizes too. Thanks @LiliDeng

Comment (sharsonia) [reply microsoft/testsuites/network/networksettings.py:187]: > > I didn't find a clear kernel version, which is after this kernel version, the case is pass, before this kernel version, the case is failed, probably it also relates with the vm size, please check.
> 
> Let me test it out with some other VM sizes too. Thanks @LiliDeng

Similar observation with D4_v3

Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108520] ethtool cpuset=/ mems_allowed=0
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108531] CPU: 3 PID: 1802 Comm: ethtool Not tainted 4.13.0-1014-azure #17-Ubuntu
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108532] Hardware name: Microsoft Corporation Virtual Machine/Virtual Machine, BIOS 090008  12/07/2018
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108533] Call Trace:
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108541]  dump_stack+0x63/0x82
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108546]  warn_alloc+0x114/0x1c0
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108551]  __vmalloc_node_range+0x1e0/0x260
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108553]  vzalloc+0x3e/0x40
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108559]  ? netvsc_device_add+0x3b6/0x7e0 [hv_netvsc]
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108561]  netvsc_device_add+0x3b6/0x7e0 [hv_netvsc]
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108565]  rndis_filter_device_add+0xd2/0x910 [hv_netvsc]
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108569]  ? kfree+0x165/0x170
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108571]  ? netvsc_teardown_gpadl+0x5e/0xc0 [hv_netvsc]
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108575]  ? call_rcu_sched+0x17/0x20
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108577]  ? netvsc_device_remove+0x94/0xc0 [hv_netvsc]
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108579]  ? rndis_filter_device_remove+0x123/0x1f0 [hv_netvsc]
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108581]  ? wait_woken+0x80/0x80
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108584]  netvsc_set_channels+0xef/0x1b0 [hv_netvsc]
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108588]  ethtool_set_channels+0x19e/0x1e0
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108590]  dev_ethtool+0x128c/0x1ee0
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108592]  ? ___slab_alloc+0x211/0x4c0
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108594]  ? page_add_file_rmap+0x58/0x170
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108596]  ? __rtnl_unlock+0x2a/0x50
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108598]  ? netdev_run_todo+0x61/0x310
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108601]  dev_ioctl+0x33e/0x540
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108603]  ? dev_ioctl+0x33e/0x540
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108607]  sock_do_ioctl+0x48/0x60
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108609]  ? sock_do_ioctl+0x48/0x60
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108611]  sock_ioctl+0x1fc/0x2d0
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108614]  do_vfs_ioctl+0xa4/0x600
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108617]  ? __do_page_fault+0x281/0x500
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108620]  SyS_ioctl+0x79/0x90
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108623]  entry_SYSCALL_64_fastpath+0x24/0xab
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108625] RIP: 0033:0x7f4d05f8df47
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108626] RSP: 002b:00007ffed64ae2f8 EFLAGS: 00000246 ORIG_RAX: 0000000000000010
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108628] RAX: ffffffffffffffda RBX: 00007ffed64ae450 RCX: 00007f4d05f8df47
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108629] RDX: 00007ffed64ae460 RSI: 0000000000008946 RDI: 0000000000000003
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108630] RBP: 00007ffed64ae320 R08: 0000000000000000 R09: 1999999999999999
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108631] R10: 000000000000053f R11: 0000000000000246 R12: 00007ffed64ae460
Jan 30 08:09:14 lisa--681-e0-n0 kernel: [  347.108632] R13: 0000000000421188 R14: 000000000041e3d6 R15: 0000000000000002

**Review by LiliDeng — DISMISSED** [review]

File: microsoft/testsuites/network/networksettings.py [hunk 2]
```diff
@@ -174,6 +174,17 @@ def validate_ringbuffer_settings_change(self, node: Node) -> None:
         priority=1,
     )
     def validate_device_channels_change(self, node: Node, log: Logger) -> None:
+        kernel_ver = node.tools[Uname].get_linux_information().kernel_version
+        if (
+            isinstance(node.os, Ubuntu)
+            and node.os.information.release <= "16.04"
+            and kernel_ver <= "4.13.15"
```
Comment (LiliDeng) [inline microsoft/testsuites/network/networksettings.py]: this is string comparison not version. if you check "4.2" <= "4.13.15", it return False

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**